src/filedialoghelper.cpp

```diff
@@ -407,9 +413,17 @@ void FileDialogHelper::saveSettings() {
     if(hiddenPlaces.isEmpty()) { // don't save "@Invalid()"
         settings.remove(QStringLiteral("HiddenPlaces"));
     }
-    else if(hiddenPlaces != settings.value(QStringLiteral("HiddenPlaces")).toStringList().toSet()) {
-        QStringList sl = hiddenPlaces.toList();
-        settings.setValue(QStringLiteral("HiddenPlaces"), sl);
+    else {
+#if (QT_VERSION >= QT_VERSION_CHECK(5,14,0))
+        QStringList hiddenPlacesList = settings.value(QStringLiteral("HiddenPlaces")).toStringList();
+        QSet<QString> hiddenPlacesSet = QSet<QString>(hiddenPlacesList.begin(), hiddenPlacesList.end());
+#else
+        QSet<QString> hiddenPlacesSet = settings.value(QStringLiteral("HiddenPlaces")).toStringList().toSet();
+#endif
+        if (hiddenPlaces != hiddenPlacesSet) {
+            QStringList sl = hiddenPlaces.values();
```

Comment:
`QSet::toList()` and `QSet::values()` are the same. If you want to use range constructors with Qt ≥ 5.14, you could write:

```
            QStringList sl (hiddenPlaces.begin(), hiddenPlaces.end());
```